### PR TITLE
MODDATAIMP Data import profile - incoming record contains 999

### PR DESCRIPTION
**Pull request by dmytrokrutii**
## Purpose
Fix a validation error when trying to import a .mrc-file with a record that has 999ff fields with this Job Profile: 
1) Modify Action Profile to remove 999 field
2) Create Instance Action Profile .

## Approach
Add 999fi validation to CreateinstanceEventHandler

## Learning
[MODDATAIMP-1001](https://folio-org.atlassian.net/browse/MODDATAIMP-1001)

### CODE REVIEW

**Review by Aliaksandr-Fedasiuk — APPROVED** [review]

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java [hunk 1]
```diff
@@ -94,6 +93,13 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
 
       Context context = EventHandlingUtil.constructContext(dataImportEventPayload.getTenant(), dataImportEventPayload.getToken(), dataImportEventPayload.getOkapiUrl());
       Record targetRecord = Json.decodeValue(payloadContext.get(EntityType.MARC_BIBLIOGRAPHIC.value()), Record.class);
+
+      if (AdditionalFieldsUtil.getValue(targetRecord, TAG_999, SUBFIELD_I).isPresent()) {
+        LOGGER.error(INSTANCE_CREATION_999_ERROR_MESSAGE);
+        future.completeExceptionally(new EventProcessingException(INSTANCE_CREATION_999_ERROR_MESSAGE));
```
[the remark this answers is not in the document]
Comment (dmytrokrutii) [reply src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java:99]: I agree, I forgot to delete it. Fixed it

**Review by JavokhirAbdullayev — APPROVED** [review]

**Review by VRohach — APPROVED** [review]

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java [hunk 2]
```diff
@@ -94,6 +93,12 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
 
       Context context = EventHandlingUtil.constructContext(dataImportEventPayload.getTenant(), dataImportEventPayload.getToken(), dataImportEventPayload.getOkapiUrl());
       Record targetRecord = Json.decodeValue(payloadContext.get(EntityType.MARC_BIBLIOGRAPHIC.value()), Record.class);
+
+      if (AdditionalFieldsUtil.getValue(targetRecord, TAG_999, SUBFIELD_I).isPresent()) {
```
Comment (RuslanLavrov) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java:97]: please, add here a check on the "acceptInstanceId" flag existence in the dataImportEventPayload context, and if it is true,
then we will allow incoming record import regardless of 999ffi field existence.

Comment (dmytrokrutii) [reply src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java]: Fixed

Comment (RuslanLavrov) [reply src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java]: Thank you!

**Review by RuslanLavrov — APPROVED** [review]

**Pull request merged**